src/transforms/babel.js

```diff
@@ -26,19 +21,81 @@ module.exports = async function(asset) {
   }
 };
 
-async function shouldTransform(asset) {
+async function getConfig(asset) {
+  // If asset is marked as an ES6 modules, this is a second pass after dependencies are extracted.
+  // Just compile modules to CommonJS.
   if (asset.isES6Module) {
-    return true;
+    return {
+      plugins: [require('babel-plugin-transform-es2015-modules-commonjs')]
+    };
+  }
+
+  if (asset.babelConfig) {
+    return asset.babelConfig;
+  }
+
+  // Attempt to generate a config
+  let config = await getEnvConfig(asset);
+
+  // If this is the app module, and there is a .babelrc, merge it into the config.
+  // Don't load .babelrc for node_modules. https://github.com/parcel-bundler/parcel/issues/13.
+  if (!asset.name.includes('/node_modules/')) {
+    let babelrc =
+      (asset.package && asset.package.babel) ||
+      (await asset.getConfig(['.babelrc', '.babelrc.js']));
+    if (babelrc) {
+      config = Object.assign({}, babelrc, {
```

Comment:
Should we merge the babelrc with the generated env config or just override if you specify a babelrc?

Reply:
I don't think that we should allow people to specify Babel transforms for `node_modules`, I think it should be exclusively `babel-preset-env` and nothing else.

Otherwise we're going to end up with all these things in the ecosystem which tell you that you need to be compiling their package in a certain way. Package authors will either say:

- You need to add this Babel plugin in order to use this package
- You cannot use these Babel plugins in this way or it will break this package

If we want to change this restriction in the future, it's easier to loosen up than to make more strict. But I think this is the right way to go

Reply:
Agreed

Reply:
@thejameskyle see check above. This only happens in the top-level app, not node_modules. In that case you could have a babelrc that we need to respect, as well as a a browserslist file which we use to enable the env transforms needed. In that case do we merge, or only use the babelrc?

Reply:
I'm not a fan of adding things to Babel configs without people actually specifying them, it's something that really annoys me with Jest